doped/utils/stenciling.py

```
@@ -0,0 +1,1122 @@
+            "by this function) should be used for its reference host cell calculation.",
+        )
+
+    _check_min_dist(oriented_new_defect_supercell, min_dist_tol)  # check distances are reasonable
+    _check_min_dist(oriented_new_bulk_supercell, bulk_min_dist_tol)
+    return oriented_new_defect_supercell, oriented_new_bulk_supercell
+
+
+def _scan_symm_ops_to_place_site_closest_to_frac_coords(
+    symm_ops: Structure | Sequence[SymmOp],
+    site: PeriodicSite,
+    target_frac_coords: np.ndarray[float] | list[float] | None = None,
```

Pylint (3.3.7)

[error] Value 'np.ndarray' is unsubscriptable

(E1136)